lib/sycamore/sycamore/query/operators/logical_operator.py

```diff
@@ -1,36 +1,33 @@
-import json
-from overrides import override
-from typing import Any, Dict, Optional
+from dataclasses import dataclass
+from typing import Dict, List, Optional
 
 from sycamore.query.logical_plan import Node
 
 
+@dataclass
+class LogicalOperatorSchemaField:
+    field_name: str
+    description: Optional[str]
+    type_hint: str
+
+
 class LogicalOperator(Node):
     """
     Logical operator class for LLM prompting.
     """
 
-    def __init__(self, node_id: str, data: Optional[Dict[Any, Any]] = None):
-        super().__init__(node_id)
-        self.data = data
-
-    @override
-    def show(self, indent=0, verbose=False):
-        print(
-            " " * indent + f"Id: {self.node_id} Operator type: {type(self).__name__} "
-            f"Data description: {self.data.get('description', 'none')}"
-        )
-        if verbose:
-            print(" " * indent + f"  {json.dumps(self.data, indent=indent)}")
-
-    @staticmethod
-    def description() -> str:
-        raise NotImplementedError
-
-    @staticmethod
-    def input_schema() -> Dict[str, Any]:
-        raise NotImplementedError
-
-    @staticmethod
-    def output_schema() -> Dict[str, Any]:
-        raise NotImplementedError
+    description: Optional[str] = None
+    """A detailed description of why this operator was chosen for this query plan."""
+
+    input: Optional[List[int]] = None
+    """A list of node IDs that this operation depends on."""
+
```

Comment:
does this need node_id as well?

Reply:
No, it gets it from the Node superclass.